### ISSUE TITLE
Can not add string to underscore template using knockout

### ISSUE THREAD

**Issue by DanielRuf** [magento/magento2#18012]
### Preconditions

1. Magento 2.4-develop

### Steps to reproduce

1. create custom theme based on Luma
2. extend Magento_Theme/frontend/web/templates/breadcrumbs.html
3. add string to breadcrumbs.html using knockout 

```
<ul class="items">
    <!-- ko i18n: 'You are here:' --><!-- /ko -->
    <% _.each(breadcrumbs, function(crumb) { %>
        <li class="item <%- crumb.name %>">
            <% if (crumb.link) { %>
            <a href="<%= crumb.link %>" title="<%- crumb.title %>"><%- crumb.label %></a>
            <% } else if (crumb.last) { %>
            <strong><%- crumb.label %></strong>
            <% } else { %>
            <%- crumb.label %>
            <% } %>
        </li>
    <% }); %>
    </ul>
```

### Expected result

1. the string "You are here:"
![image](https://user-images.githubusercontent.com/51680850/71517608-8b55be80-28b7-11ea-9f7b-6df06542dd70.png)

### Actual result

1. No string shown
![image](https://user-images.githubusercontent.com/51680850/71517594-7711c180-28b7-11ea-8c40-e2c8afec52d4.png)

**Comment by magento-engcom-team** [comment]
Hi @DanielRuf. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@DanielRuf do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by DanielRuf** [comment]
It seems only `<%= jQuery.mage.__('You are here:') %>` works. `<%= $.mage.__('You are here:') %>` does not.

**Comment by ghost** [comment]
Hi @DanielRuf I'm not able to reproduce by following steps you described, in my case i ovveride 
breadcrumbs.phtml with this code 
```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

// @codingStandardsIgnoreFile

?>
<?php if ($crumbs && is_array($crumbs)) : ?>
    <div class="breadcrumbs">
        <ul class="items">
            <!-- ko i18n: 'You are here:' --><!-- /ko -->
            <?php foreach ($crumbs as $crumbName => $crumbInfo) : ?>
                <li class="item <?= /* @escapeNotVerified */ $crumbName ?>">
                    <?php if ($crumbInfo['link']) : ?>
                        <a href="<?= /* @escapeNotVerified */ $crumbInfo['link'] ?>" title="<?= $block->escapeHtml($crumbInfo['title']) ?>"><?= $block->escapeHtml($crumbInfo['label']) ?></a>
                    <?php elseif ($crumbInfo['last']) : ?>
                        <strong><?= $block->escapeHtml($crumbInfo['label']) ?></strong>
                    <?php else: ?>
                        <?= $block->escapeHtml($crumbInfo['label']) ?>
                    <?php endif; ?>
                </li>
            <?php endforeach; ?>
        </ul>
    </div>
<?php endif; ?>

```

![selection_072](https://user-images.githubusercontent.com/40756439/45353907-eaa99580-b5c4-11e8-8836-de29b3929af9.png)

**Comment by DanielRuf** [comment]
> breadcrumbs.phtml with this code

This is wrong, I mean breadcrumbs.html, please read again.

**Comment by DanielRuf** [comment]
> Magento_Theme/frontend/web/templates/breadcrumbs.html

**Comment by DanielRuf** [comment]
```
<ul class="items">
    <!-- ko i18n: 'You are here:' --><!-- /ko -->
    <% _.each(breadcrumbs, function(crumb) { %>
        <li class="item <%- crumb.name %>">
            <% if (crumb.link) { %>
            <a href="<%= crumb.link %>" title="<%- crumb.title %>"><%- crumb.label %></a>
            <% } else if (crumb.last) { %>
            <strong><%- crumb.label %></strong>
            <% } else { %>
            <%- crumb.label %>
            <% } %>
        </li>
    <% }); %>
    </ul>
```

**Comment by DanielRuf** [comment]
https://github.com/magento/magento2/issues/18012#issuecomment-420192906

**Comment by ghost** [comment]
@DanielRuf  , thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by jayankaghosh** [comment]
Since `Magento_Theme/frontend/web/templates/breadcrumbs.html` is an underscore template, wouldn't knockout bindings not work anyhow here? 

And `<%= $.mage.__('You are here:') %>` doesn't work because the variable `$` wasn't passed to the underscore templating engine while parsing the breadcrumbs template.

`<%= jQuery.mage.__('You are here:') %>` works since the variable `jQuery` is defined in the `window` object.

Am I missing something here?

**Comment by DanielRuf** [comment]
> And `<%= $.mage.__('You are here:') %>` doesn't work because the variable `$` wasn't passed to the underscore templating engine while parsing the breadcrumbs template.
> 
> `<%= jQuery.mage.__('You are here:') %>` works since the variable `jQuery` is defined in the `window` object.
> 
> Am I missing something here?

Exactly.

> Since `Magento_Theme/frontend/web/templates/breadcrumbs.html` is an underscore template, wouldn't knockout bindings not work anyhow here?

Not sure how it should look like but the docs are not clear here.
See https://devdocs.magento.com/guides/v2.0/frontend-dev-guide/translations/translate_theory.html

**Comment by jayankaghosh** [comment]
What could be done is, under the underscore object, an i18n method could be introduced, so that translations would work if we wrote something like `<%= _.i18n('You are here:') %>`

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Charlie 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30107` were created

**Issue Available**: @engcom-Charlie, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by slavvka** [comment]
Hi @DanielRuf. Thank you for your report. 
The issue has been fixed in magento/magento2#26435 by @sergiy-v in 2.4-develop branch
Related commit(s):
 - [819c3f255224d7e2d04e90bcd47bb410a2eee23b](https://github.com/magento/magento2/commit/819c3f255224d7e2d04e90bcd47bb410a2eee23b)
 - [064ee0ef6d59ca3fb27802d74609377b4816e869](https://github.com/magento/magento2/commit/064ee0ef6d59ca3fb27802d74609377b4816e869)
 - [3a19f179309633f3c12ee8e039e5dd5b2d513624](https://github.com/magento/magento2/commit/3a19f179309633f3c12ee8e039e5dd5b2d513624)
 - [c32b64377fc06771f12e74b4add2f32c8fa0a555](https://github.com/magento/magento2/commit/c32b64377fc06771f12e74b4add2f32c8fa0a555)
 - [e26bbc17188b94b6c8bac770d3d4586576f28267](https://github.com/magento/magento2/commit/e26bbc17188b94b6c8bac770d3d4586576f28267)
 - [69d07c13b394bdf367048ba25dbf922f7cc4f1b6](https://github.com/magento/magento2/commit/69d07c13b394bdf367048ba25dbf922f7cc4f1b6)
 - [58adcc59e675430e890ab5d18f675e90715646a1](https://github.com/magento/magento2/commit/58adcc59e675430e890ab5d18f675e90715646a1)

The fix will be available with the upcoming 2.4.0 release.
